sympy/parsing/c/c_parser.py

```diff
@@ -735,6 +738,64 @@ def transform_decl_stmt(self, node):
 
             return statement
 
+        def transform_paren_expr(self, node):
+            """To handle outermost parenthesis of a well formed rhs"""
+            return self.transform(next(node.get_children()))
+
+        def transform_unary_operator(self, node):
+            """Transformation function for handling unary operators
+
+            Returns
+            =======
+
+            unary_expression: Codegen AST node
+                    simplified unary expression represented as Codegen AST
+
+            Raises
+            ======
+
+            NotImplementedError if post increment/decrement operator
+            is given or other than supported operators given in the list
+
+            """
+            # supported operators list
+            operators_list = ['+', '-', '++', '--', '!']
+            tokens = [token for token in node.get_tokens()]
+
+            # it can be either pre increment/decrement or any other operator from the list
+            if tokens[0].spelling in operators_list:
+                child = self.transform(next(node.get_children()))
+                # (decl_ref) e.g.; int a = ++b; or simply ++b;
+                if isinstance(child, str):
+                    if tokens[0].spelling == '+':
+                        return Symbol(child)
+                    if tokens[0].spelling == '-':
+                        return Mul(Symbol(child), -1)
+                    if tokens[0].spelling == '++':
+                        return Assignment(Symbol(child),
+                            Add(Symbol(child), 1))
+                    if tokens[0].spelling == '--':
+                        return Assignment(Symbol(child),
+                            Add(Symbol(child), -1))
+                    if tokens[0].spelling == '!':
+                        return Not(Symbol(child))
+                # e.g.; int a = -1; or int b = -(1 + 2);
+                else:
+                    if tokens[0].spelling == '+':
+                        return child
+                    if tokens[0].spelling == '-':
+                        return Mul(child, -1)
+                    if tokens[0].spelling == '!':
+                        return Not(sympify(bool(child)))
+
+            # it can  be either post increment/decrement
+            # since variable name is obtained in token[1].spelling
+            elif tokens[1].spelling in ['++', '--']:
```

Comment:
It is fairly easy to convert single statement `a++;` or `a--;`, the real problem arises when it is used with `var_decl` :)
Actually, the difficulty that I am currently facing in implementation is:
#### Example:
`template.cpp`:
<pre>
void func(){
	int a = 0;
	int b = a++;
}
</pre>

In case of pre increment/decrement, it is safe to assume `int b = a = a + 1;`. Hence, we can convert unary operator part (` a++;` of `int b = a++;`) to return `Assignment(Symbol('a'), Add(Symbol('a'), 1))`

But, in this case(i.e in post increment/decrement), it would be similar to `int b = a; a = a + 1;` (return `Symbol('a')` from `transform_unary_operator()` and then include another node `Assignment(Symbol('a'), Add(Symbol('a'), 1))` to the `_py_nodes`)

Now, how to do something like this?

One possible method can be setting an instance variable(a flag maybe) to denote an operation to be performed next (i.e. use that flag in `transform_var_decl()` to add a node `Assignment(Symbol('a'), Add(Symbol('a'), 1))` to the `_py_nodes` after variable declaration is over). But this looks more like a workaround and I don't want to have workarounds everywhere :)

Please guide me if this is a legit way or suggest any other way, which is better!

Reply:
You should use the `PostDecrement` node in https://github.com/sympy/sympy/blob/8203f62e5e606de01830fd452e75484be4af922e/sympy/codegen/cnodes.py#L78

Although, I do like the current method as it ensures the expression is supported by all the printers while the `cnodes` nodes are not supported by all the printers. But, I think it will be better if the nodes and the printers handle the scenario instead of the parser.

Reply:
Seems a better approach, working on it!